src/unix/threadpsx.cpp

```diff
@@ -879,6 +880,15 @@ void *wxThreadInternal::PthreadStart(wxThread *thread)
                    wxT("Thread %p about to enter its Entry()."),
                    THR_ID(pthread));
 
+        if ( wxSystemOptions::IsFalse("catch-unhandled-exceptions") )
+        {
+            pthread->m_exitcode = thread->Entry();
+
+            wxLogTrace(TRACE_THREADS,
+                       wxT("Thread %p Entry() returned %lu."),
+                       THR_ID(pthread), wxPtrToUInt(pthread->m_exitcode));
```

Comment:
I know this is a much smaller piece of code, but it would be more consistent to move it into a function too. Or maybe we could use the helper taking lambda, see the other comment.